### PR TITLE
KATI-108_Patient-Pending-Appt-Cancel-Function

### PR DESCRIPTION
**Pull request by vtnlinh95**

### CODE REVIEW

**Comment by hieumai** [comment]
Please fix the commits ( why is there a ticket with "KATI - 90", and the one with "KATI-108" contain only minor changes ). Also please resolve conflicts so I can review.

**Comment by vtnlinh95** [comment]
@hieumai fixed

File: app/src/main/java/com/kms/cura/model/Settings.java [hunk 1]
```diff
@@ -4,7 +4,8 @@
  * Created by linhtnvo on 6/6/2016.
  */
 public class Settings {
-    public static final String SERVER_URL = "http://192.168.74.141:8080/cura/cura_server";
+    //public static final String SERVER_URL = "http://192.168.74.141:8080/cura/cura_server";
+    public static final String SERVER_URL = "http://10.0.3.2:8080/com.kms.cura_server/cura_server";
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/model/Settings.java]: Local server url here :-1:

File: app/src/main/java/com/kms/cura/view/fragment/PatientAppointmentListTabFragment.java [hunk 2]
```diff
@@ -92,6 +96,7 @@ protected void onPostExecute(Void aVoid) {
                 task.execute();
             }
         });
+        EventBroker.getInstance().register(this, EventConstant.UPDATE_APPT_PATIENT_LIST);
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/fragment/PatientAppointmentListTabFragment.java:99]: There should be a place that unregister event for this fragment

File: app/src/main/java/com/kms/cura/view/fragment/PatientAppointmentListTabFragment.java [hunk 3]
```diff
@@ -150,7 +155,23 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
...
+                toDetails.putExtra(APPT_POSITION, appts.indexOf(entity));
+                startActivity(toDetails);
+            }
+        });
     }
 
 
+    @Override
+    public void handleEvent(String event, Object data) {
+        apptsList.clear();
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/fragment/PatientAppointmentListTabFragment.java]: Should check for event type even if you registered 1 event for this

File: com.kms.cura.dal/src/com/kms/cura/dal/AppointmentDAL.java [hunk 4]
```diff
@@ -42,4 +42,14 @@ public static AppointmentDAL getInstance() {
 			dbh.closeConnection();
 		}
 	}
+	
+	public List<AppointmentEntity> updateAppointment(AppointmentEntity entity) throws ClassNotFoundException, SQLException {
+		AppointmentDatabaseHelper dbh = new AppointmentDatabaseHelper();
+		try{
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/AppointmentDAL.java]: re-format this file

File: com.kms.cura.dal/src/com/kms/cura/dal/database/AppointmentDatabaseHelper.java [hunk 5]
```diff
@@ -170,5 +176,33 @@ private void createAppointment(AppointmentEntity entity) throws SQLException {
...
+			return patientAppts;
+		}
+		 catch (SQLException e) {
+				if (con != null) {
+					System.err.print("Transaction is being rolled back");
+					con.rollback();
+				}
+				throw e;
+		 }
+		finally{
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/AppointmentDatabaseHelper.java]: re-format this file

**Comment by hieumai** [comment]
OK from me. @huuduy216  and @toannguyen71195  please start verifying this

**Comment by hieumai** [comment]
Looks OK. @vtnlinh95  please resolve conflicts and squash them to 1 commits

**Pull request merged**